### PR TITLE
hub: Add handling for resending verification code

### PR DESCRIPTION
**Pull request by backspace**
If `POST /api/email-card-drop-requests` happens with an EOA and hashed email that already exists but `claimed_at` is absent, this updates the verification code and sends another email.

### CODE REVIEW

File: packages/hub/routes/email-card-drop-requests.ts [hunk 1]
```diff
@@ -140,7 +140,32 @@ export default class EmailCardDropRequestsRoute {
...
+    let unclaimedButExisting = await this.emailCardDropRequestQueries.query({
+      emailHash: normalizedEmailHash,
+      ownerAddress: ctx.state.userAddress,
+    });
+
+    if (unclaimedButExisting.length > 0) {
+      let existingRequest = unclaimedButExisting[0];
+      let updatedEmailCardDropRequest = await this.emailCardDropRequestQueries.updateVerificationCode(
+        existingRequest.id,
+        generateVerificationCode()
```
Comment (Aierie) [inline packages/hub/routes/email-card-drop-requests.ts:152]: Why do we need to update the verification code?

File: packages/hub/routes/email-card-drop-requests.ts [hunk 2]
```diff
@@ -140,7 +140,32 @@ export default class EmailCardDropRequestsRoute {
...
+      );
+
+      await this.workerClient.addJob('send-email-card-drop-verification', {
+        id: updatedEmailCardDropRequest.id,
```
Comment (Aierie) [inline packages/hub/routes/email-card-drop-requests.ts:156]: Also, not related to this PR but wondering if this mutability of the verification code is why we don't simply pass the EOA + verification code into this job?

**Review by Aierie — APPROVED** [review]
This ran nicely for me locally.

File: packages/hub/routes/email-card-drop-requests.ts (hunk 1 above)
Comment (backspace) [reply packages/hub/routes/email-card-drop-requests.ts:152]: hmm… I guess I’m just following the convention, I’m not sure why the convention exists though 🤔

File: packages/hub/routes/email-card-drop-requests.ts (hunk 2 above)
Comment (backspace) [reply packages/hub/routes/email-card-drop-requests.ts:156]: I debated including it, do you think it’s preferable?

**Pull request merged**